app/src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -177,4 +198,63 @@ private void loadCaCert() {
             }
         }
     }
+
+    /**
+     * Set webview proxy and sites that are not retrieved using proxy.
+     * Compatible with KitKat or higher android version.
+     * Returns boolean if successful.
+     * Source: https://stackoverflow.com/a/26781539
+     */
+    public static boolean setWebViewProxy(Context appContext, String host, int port, String exclusionList) {
+        Properties properties = System.getProperties();
+        properties.setProperty("http.proxyHost", host);
+        properties.setProperty("http.proxyPort", port + "");
+        properties.setProperty("https.proxyHost", host);
+        properties.setProperty("https.proxyPort", port + "");
+        properties.setProperty("http.nonProxyHosts", exclusionList);
+        properties.setProperty("https.nonProxyHosts", exclusionList);
+
+        if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
+            // Not supported on android version lower than KitKat.
+            return false;
+        }
+
+        try {
+            Class applictionCls = Class.forName("android.app.Application");
+            Field loadedApkField = applictionCls.getDeclaredField("mLoadedApk");
+            loadedApkField.setAccessible(true);
```

Comment:
Is there no non-magical non-relfection way to do this? Where did you find the guide to do this?

Reply:
Unfortunately no. After doing deeper research, I am convinced this can only be done using reflection. Source was https://stackoverflow.com/a/26781539

Reply:
Thats correct. There are also workarounds for older Android versions around, but I think we should not fix the issue with the web UI not loading for that old android versions. It should be rare that a user uses proxy and a very old android version and the fix for ICS would add more complexity we carry with us.